saw-core-coq/src/Verifier/SAW/Translation/Coq/SAWModule.hs

```diff
@@ -93,22 +93,22 @@ translateDataType (DataType {..}) =
     translateNamed name = do
       let inductiveName = name
       (inductiveParameters, inductiveIndices) <-
-        liftTermTranslationMonad $ do
-        ps <- TermTranslation.translateParams dtParams
-        ixs <- TermTranslation.translateParams dtIndices
+        liftTermTranslationMonad $
+        TermTranslation.translateParams dtParams $ \ps ->
+        TermTranslation.translateParams dtIndices $ \ixs ->
         -- Translating the indices of a data type should never yield
         -- Inhabited constraints, so the result of calling
         -- `translateParams dtIndices` above should only return Binders and not
         -- any ImplicitBinders. Moreover, `translateParams` always returns
         -- Binders where the second field is `Just t`, where `t` is the type.
-        let errorBecause msg = error $ "translateDataType.translateNamed: " ++ msg
+        let errorBecause msg = error $ "translateDataType.translateNamed: " ++ msg in
         let bs = map (\case Coq.Binder s (Just t) ->
                               Coq.PiBinder (Just s) t
                             Coq.Binder _ Nothing ->
                               errorBecause "encountered a Binder without a Type"
                             Coq.ImplicitBinder{} ->
                               errorBecause "encountered an implicit binder")
-                     ixs
+                     ixs in
         return (ps, bs)
```

Comment:
I'm a bit confused by this change, it looks a bit awkward, and I don't see the benefits?

Reply:
Ignore this, same thing as the other one. I know realize I was thinking of it as it the lines ended with `>>= \ ps` but it's `$ \ ps`, very different. :-)

Reply:
The idea is that the computations in those lambdas are run in a different variable binding context than the surrounding computation containing the `translateParams` call. Having those be lambdas makes this fact explicit, so it's easy to see which things are being translated in the extended context vs in the original context. That is, it makes the change in variable-binding context part of the API, so it is apparent to the caller. With the previous state-based approach, some sub-function could modify the state and add more variable bindings in an implicit way, and that fact was only represented in the comments on the sub-function.

State differently, we don't _want_ it to be easy to just bind variables willy nilly. We _want_ it to be a little awkward, to force users of the API to be very explicit about where they are being bound.

Does this make sense?